pkg/handler/collector/s3/messaging/sqs.go

```
@@ -0,0 +1,161 @@
+	Records []SqsRecord `json:"Records"`
+}
+
+func (m *SqsMessage) GetEvent() (EventName, error) {
+	if len(m.Records) == 0 {
+		return "", fmt.Errorf("error getting event from message %s", m)
+	}
+
+	if m.Records[0].EventName == "ObjectCreated:Put" {
+		return PUT, nil
+	}
+
+	return "", nil
+}
+
+func (m *SqsMessage) GetBucket() (string, error) {
+	if len(m.Records) == 0 {
+		return "", fmt.Errorf("error getting bucket from message %s", m)
+	}
+	return m.Records[0].S3.Bucket.Name, nil
+}
+
+func (m *SqsMessage) GetItem() (string, error) {
+	if len(m.Records) == 0 {
+		return "", fmt.Errorf("error getting item from message %s", m)
+	}
+	return m.Records[0].S3.Object.Key, nil
+}
+
+func NewSqsProvider(mpConfig MessageProviderConfig) (SqsProvider, error) {
+	sqsHostname := mpConfig.Host
+	sqsPort := mpConfig.Port
+	sqsQueue := mpConfig.Queue
+	sqsProvider := SqsProvider{}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return SqsProvider{}, fmt.Errorf("error loading AWS SDK config: %w", err)
+	}
+
+	client := sqs.NewFromConfig(cfg, func(o *sqs.Options) {
+		o.BaseEndpoint = aws.String(fmt.Sprintf("http://%s:%s", sqsHostname, sqsPort))
+		o.Region = mpConfig.Region
+	})
+
+	sqsProvider.client = client
+	sqsProvider.hostname = sqsHostname
+	sqsProvider.port = sqsPort
+	sqsProvider.queue = sqsQueue
+
+	return sqsProvider, nil
+}
+
+func (s *SqsProvider) ReceiveMessage(ctx context.Context) (Message, error) {
+	logger := logging.FromContext(ctx)
+
+	addr := fmt.Sprintf("http://%s:%s/000000000000/%s", s.hostname, s.port, s.queue)
+
+	// Receive messages from the queue
+	receiveInput := &sqs.ReceiveMessageInput{
+		QueueUrl:            &addr,
+		MaxNumberOfMessages: 1,
+		WaitTimeSeconds:     10,
+	}
+
+	select {
+	case <-ctx.Done():
+		return &SqsMessage{}, ctx.Err()
+	default:
+		receiveOutput, err := s.client.ReceiveMessage(ctx, receiveInput)
+		if err != nil {
+			fmt.Printf("error receiving message, skipping: %v\n", err)
+			//continue
```

Should return an error here.